tests/Constraints/ConstTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the JsonSchema package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JsonSchema\Tests\Constraints;
+
+class ConstTest extends BaseTestCase
+{
+    protected $schemaSpec = 'http://json-schema.org/draft-06/schema#';
+    protected $validateSchema = true;
+
+    public function getInvalidTests()
+    {
+        return array(
+            array(
+                '{"value":"foo"}',
+                '{
+                  "type":"object",
+                  "properties":{
+                    "value":{"type":"string","const":"bar"}
+                  },
+                  "additionalProperties":false
+                }'
+            ),
+            array(
+                '{"value":5}',
+                '{
+                  "type":"object",
+                  "properties":{
+                    "value":{"type":"integer","const":6}
+                  },
+                  "additionalProperties":false
+                }'
+            ),
+            array(
+                '{"value":false}',
+                '{
+                  "type":"object",
+                  "properties":{
+                    "value":{"type":"boolean","const":true}
+                  },
+                  "additionalProperties":false
+                }'
+            ),
+        );
+    }
+
+    public function getValidTests()
+    {
+        return array(
+            array(
+                '{"value":"bar"}',
+                '{
+                  "type":"object",
+                  "properties":{
+                    "value":{"type":"string","const":"bar"}
```

Comment:
I'm a little curious about the potential use cases covered by this implementation. Ten minutes ago I had never heard of the `const` keyword, but from what I've read since then its main utility comes from interacting with the `$data` concept proposed in v5. Given that we don't currently support `$data` or relative JSON pointers, how much use is there in supporting a keyword that guarantees that an input value is precisely equal to some other value?

Reply:
I did not find any mention of the `$data` concept in the recent spec v7. From my understanding, `const` is presumably nothing more than a more expressive alternative to a one-element `enum`:

>**6.1.3. const**
>The value of this keyword MAY be of any type, including null.
>
>An instance validates successfully against this keyword if its value is equal to the value of the keyword.

My use case for `const` is when using a "constant" property as a type discriminator when using polymorphic types:

```
{
  "$oneOf": [
    {
      "properties": {
        "paymentType": {"type": "string", "const": "directDebit"},
        "iban": {"type": ...}
       }
    },
    {
      "properties": {
        "paymentType": {"type: "string", "const": "creditCard"},
        "cardNumber": {...}
      }
  ]
}
```

In this case, documents like `{"payment": {"paymentType": "directDebit", "iban": "DE1234567890"}}` and `{"payment": {"paymentType": "creditCard", "cardNumber": "123456789"}}` would match the schema, while something like `{"payment": {"paymentType": "creditCard", "iban": "DE1234567890"}}` would not.

Reply:
I see. Okay. 👍